### PR TITLE
Add az-schema-type-and-format rule with tests and doc

### PR DESCRIPTION
**Pull request by mikekistler**
This PR adds a rule to check that the type and format combination for a schema is well-defined.

### CODE REVIEW

File: functions/schema-type-and-format.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  const errors = [];
+
+  const stringFormats = [
+    // OAS-defined formats
+    'byte', 'binary', 'date', 'date-time', 'password',
+    // Additional formats recognized by autorest
+    'char', 'time', 'date-time-rfc1123', 'duration', 'uuid', 'base64url', 'url',
+    'odata-query', 'certificate',
```
Comment (jianyexi) [inline functions/schema-type-and-format.js:19]: we want to support 'uri' as well, see https://github.com/Azure/azure-openapi-validator/issues/345

File: functions/schema-type-and-format.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  } else if (schema.properties && typeof schema.properties === 'object') {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [key, value] of Object.entries(schema.properties)) {
+      errors.push(
+        ...checkTypeAndFormat(value, options, { path: [...path, 'properties', key] }),
+      );
+    }
+  }
+
```
Comment (jianyexi) [inline functions/schema-type-and-format.js:74]: I think we also need to check the schema inside 'allOf'

**Comment by jianyexi** [comment]
BTW, why we still adding rule here , do you change to 'azure-openapi-validator' , I think I can enable a staging pipeline for all the spectral rules this week

File: functions/schema-type-and-format.js (hunk 2 above)
Comment (mikekistler) [reply functions/schema-type-and-format.js:74]: Fixed.

**Comment by mikekistler** [comment]
Agreed that we should consolidate on 'azure-openapi-validator' once the Spectral rules are merged there. I think this particular rule (or one very much like it) will replace the 'ValidFormats' rule.

**Pull request merged**